test: Update the new ZTEST APIs tests/net/socket/af_packet

hunterhu3000 opened the pull request:
TEST=twister -T tests/net/socket/af_packet

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

nashif commented on tests/net/socket/af_packet/src/main.c:
```diff
@@ -215,7 +215,7 @@ static void __test_packet_sockets(int *sock1, int *sock2)
 #define UDP_HDR_SIZE 8
 #define HDR_SIZE (IP_HDR_SIZE + UDP_HDR_SIZE)
-static void test_raw_packet_sockets(void)
+ZTEST(socket_packet_test_suite, test_raw_packet_sockets)
```
no need for _test_suite here.

hunterhu3000 committed:
tests: net: socket: move to new ztest API

Move net tests to use new ztest API
TEST=twister -T tests/net/socket/af_packet

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

nashif commented:
btw, your commit message body is not enough, you need to be descriptive in the body and not just have a command line of twister. Please read the contribution guidelines.

hunterhu3000 commented:
> 

Thank you for your comment.
I think the comment in the PR is not up to date. Maybe I should manually update the title and the comment in the PR.
My original commit info is like this 
commit 611866f6a1798b05cec01f35014501d7ddf1c617 (HEAD -> net_socket_af_packet, origin/net_socket_af_packet)
Author: Hu Zhenyu <zhenyu.hu@intel.com>
Date:   Tue Jul 26 09:05:23 2022 +0800

    tests: net: socket: move to new ztest API

    Move net tests to use new ztest API
    TEST=twister -T tests/net/socket/af_packet

    Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

Do you think it is OK? If yes, I'll follow this to submit the rest. Thanks.
